tensorflow/python/estimator/inputs/pandas_io.py

```diff
@@ -87,7 +103,12 @@ def pandas_input_fn(x,
     if not np.array_equal(x.index, y.index):
       raise ValueError('Index for x and y are mismatched.\nIndex for x: %s\n'
                        'Index for y: %s\n' % (x.index, y.index))
-    x[target_column] = y
+    if isinstance(y, pd.DataFrame):
+      y_columns = [(column, _get_unique_target_key(x, column)) for column in list(y)]
```

Comment:
The _get_unique_target_key trick can work if y is single object not a list. Here, it is still possible to have conflicted keys. Say the keys for x is ['a'], the keys for y is ['a', 'a_n'].

Reply:
I see. So I need to have `_get_unique_target_key` check that the unique key isn't contained in `x` or in `y`.